app/Signup/page.tsx

```diff
@@ -1,9 +1,146 @@
-import styles from "./page.module.css"
+'use client';
+import styles from "./page.module.css";
+import { useState} from "react";
 
 export default function Page() {
+
+    const [user, setUser] = useState({
+        firstName: "",
+        lastName: "",
+        userName: "",
+        houseName: "",
+        userEmail: "",
+        userPassword: "" 
+    });
+    const [password, setPassword] = useState("");
+    const [rePassword, setRePassword] = useState("");
+    const [houseError, setHouseError] = useState(false);
+    const [emailError, setEmailError] = useState(false);
+    const [userPasswordError, setUserPasswordError] = useState(false);
+    const [userRePasswordError, setUserRePasswordError] = useState(false);
+    const hasError = houseError || emailError || userPasswordError || userRePasswordError;
+
+    const validateHouseName = (e) => {
+        const alphaNumericCheck = /^[a-z\d\-_\s]+$/i;
+        if ( !alphaNumericCheck.test(e.target.value)) {
+            setHouseError(true);
```

Comment:
put a return statement below line 26. So you can use a "guard clause" and not have to use that else statement. In functions if you use the "return" keyword it'll "return whatever you put after it" and end execution of that function. 

If your return statement is empty, so just using the return keyword it'll return null fyi.